src/TestcafeGherkinBootstrapper.js

```diff
@@ -140,20 +140,28 @@ module.exports = class TestcafeGherkinBootstrapper extends TestcafeBootstrapper
   }
 
   _shouldRunScenario(scenario) {
-    if (!this.tags.length) {
-      return true;
-    }
+    return this._scenarioHasAnyOfTheTags(scenario, this._getIncludingTags(this.tags)) &&
+      this._scenarioLacksTags(scenario, this._getExcludingTags(this.tags));
+  }
+
+  _getIncludingTags(tags) {
+    return tags.filter(tag => !tag.startsWith('~'));
+  }
 
-    return this._scenarioHasAnyOfTheTags(scenario, this.tags);
+  _getExcludingTags(tags) {
+    return tags
+      .filter(tag => tag.startsWith('~'))
+      .map(tag => tag.slice(1)); // Remove tilde
   }
 
   _scenarioHasAnyOfTheTags(scenario, tags) {
-    for (let i = 0; i < scenario.tags.length; i++) {
-      if (tags.indexOf(scenario.tags[i].name) > -1) {
-        return true;
-      }
+    if (!tags.length) {
+      return true;
     }
+    return tags.some(tag => scenario.tags.map(t => t.name).includes(tag));
+  }
 
-    return false;
+  _scenarioLacksTags(scenario, tags) {
+    return !tags.some(tag => scenario.tags.map(t => t.name).includes(tag));
```

Comment:
There is code duplication here.

```suggestion
    return !this._scenarioHasAnyOfTheTags(scenario, tags);
```